File: e2e/factories/companyStripeAccounts.ts
import { db } from "@test/db";
import { companiesFactory } from "@test/factories/companies";
import { companyStripeAccounts } from "@/db/schema";

export const companyStripeAccountsFactory = {
  create: async (overrides: Partial<typeof companyStripeAccounts.$inferInsert> = {}) => {
    const insertedStripeAccount = await db
      .insert(companyStripeAccounts)
      .values({
        companyId: overrides.companyId || (await companiesFactory.create()).company.id,
        status: "ready",
        setupIntentId: "seti_1LS2aCFSsGLfTpetJF5ZbTzr",
        bankAccountLastFour: "4242",
        ...overrides,
      })
      .returning();

    return { stripeAccount: insertedStripeAccount[0] };
  },

  createInitial: async (overrides: Partial<typeof companyStripeAccounts.$inferInsert> = {}) =>
    companyStripeAccountsFactory.create({
      status: "initial",
      bankAccountLastFour: null,
      ...overrides,
    }),

  createActionRequired: async (overrides: Partial<typeof companyStripeAccounts.$inferInsert> = {}) =>
    companyStripeAccountsFactory.create({
      status: "action_required",
      ...overrides,
    }),

  createProcessing: async (overrides: Partial<typeof companyStripeAccounts.$inferInsert> = {}) =>
    companyStripeAccountsFactory.create({
      status: "processing",
      ...overrides,
    }),
};

File: apps/rails/spec/factories/company_stripe_accounts.rb
# frozen_string_literal: true

FactoryBot.define do
  factory :company_stripe_account do
    company

    status { "ready" }
    setup_intent_id { "seti_1LS2aCFSsGLfTpetJF5ZbTzr" }
    bank_account_last_four { "4242" }

    trait :initial do
      status { "initial" }
      bank_account_last_four { nil }
    end

    trait :action_required do
      status { "action_required" }
    end
  end
end
